src/integrations/PinterestTag/browser.js

```diff
@@ -73,6 +74,43 @@ export default class PinterestTag {
   }
   /* utility functions --- Ends here ---  */
 
+  generateLdpObject(message) {
+    if (!message) {
+      const { userTraits } = this.analytics;
+      const optOutType = userTraits?.optOutType;
+      const state =
+        userTraits?.state ||
+        userTraits?.State ||
+        userTraits?.address?.state ||
+        userTraits?.address?.State ||
+        '';
+      const country =
+        userTraits?.country ||
+        userTraits?.Country ||
+        userTraits?.address?.country ||
+        userTraits?.address?.Country ||
+        '';
+      return {
+        opt_out_type: optOutType,
+        st: sha256(state),
+        country: sha256(country),
+      };
+    }
+
+    const optOutType = message.context.traits?.optOutType || message.properties?.optOutType;
+    const { state, country } = getDefinedTraits(message);
+    return {
+      opt_out_type: optOutType,
+      st: sha256(state || ''),
+      country: sha256(country || ''),
+    };
```

Comment:
If optOutType is undefined, do we need to set it?

Reply:
code updated. Setting `optOutType` to empty strings If it is undefined.